staging/src/k8s.io/apimachinery/pkg/runtime/serializer/json/json.go

```diff
@@ -128,20 +131,24 @@ func (s *Serializer) Decode(originalData []byte, gvk *schema.GroupVersionKind, i
 		return nil, nil, err
 	}
 
-	if gvk != nil {
-		// apply kind and version defaulting from provided default
+	// apply kind and version defaulting from provided default
+	completeGVK := func (actual *schema.GroupVersionKind, defaultGVK schema.GroupVersionKind) {
```

Comment:
just noticed that this is just a `schema.GroupVersionKind`. So better write this functionally and move the helper func out:

```
func gvkWithDefaults(gvk schema.GroupVersionKind, defaults) schema.GroupVersionKind
```